server/server.go

```diff
@@ -222,6 +223,19 @@ func (s *Server) handleGetMh(w lookupResponseWriter, r *http.Request) {
 	mh := w.Key()
 	log := logger.With("multihash", mh)
 
+	dmh, err := multihash.Decode(mh)
+	if err != nil {
+		logger.Errorw("Failed to decode multihash")
+		http.Error(w, "", http.StatusInternalServerError)
+		return
+	}
+
+	// handle only double hashed multihashes
+	if dmh.Code != multihash.DBL_SHA2_256 {
+		http.Error(w, "", http.StatusNotFound)
```

Comment:
I think it should handle both transparently. no?

Reply:
Currently dhstore answers to double hashed requests and dhfind to regular ones.